### PR TITLE
Fourier features

### LINKED ISSUE #121: [Suggestion for "Painting"] Use of Fourier Features?
Hello,

On the page titled ["Image Painting"](https://cs.stanford.edu/people/karpathy/convnetjs/demo/image_regression.html) I wanted to suggest passing the (x, y) points through a Fourier feature mapping. The details on this are described here in this [2020 NeurIPS paper](https://bmild.github.io/fourfeat/index.html) by Google Research ([including example code](https://github.com/tancik/fourier-feature-networks/blob/master/Demo.ipynb)). 

Admittedly I'm not sure if this would be out of scope for an intro example for this library, but it would be very cool to see the neural net be able to paint the cat clearly. 

Btw thanks for sharing this great work! I originally found this repo when I was in undergrad, it was part of what got me excited about computer vision :)

### PR DESCRIPTION
**Pull request by UPstartDeveloper**
Closes #121. It's not perfect but I'd love to get folks' feedback on the current state of this code on this branch!

**Change Summary**:
1. I added a new layer class called `FourierFeatureLayer`, and demo its usage by including it in the default neural network defined in the `image_regression.js` file
2. I added my contact link in `image_regression.html` (in addition to keeping @karpathy 's Twitter handle link). Happy to keep working on this example and/or engage in further discussion about it!

**Known limitations:**
1. For some reason, passing a value greater than `0` for `gaussian_mapping_scale` causes the loss of the model to plateau. For now I've set it so that it defaults to use using sinusoidal functions in the mapping (please the [paper](https://bmild.github.io/fourfeat/index.html) I'm working off of in case that sentence doesn't make sense).
3. the addition of the Fourier Feature layer in `image_regression.js` causes the wall clock time for training to increase, and increases proportional to how high `gaussian_mapping_scale`.

### CODE REVIEW

**Comment by 371148606** [comment]
您好，我是薛文毅，已收到您的邮件，一会查收，谢谢！！
